### ISSUE TITLE
Site 199-Willow Grove SAB - Account Balance issue

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#461]
In GitLab by @kdjstudios on Sep 7, 2017, 08:41

**Submitted by:** "Cindy Pesa" <cindy.pesa@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2017-09-07-79368/conversation

**Server:** Internal

**Client/Site:** Willow Grove

**Account:**  National Truck Protection

**Issue:** 

Account - National Truck Protection - is showing a balance owed as 6440.00 and the true balance is accurate in the account ledger 4090.00. This needs to be corrected.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 7, 2017, 08:44

@smasih @tim.traylor 

I believe I checked the account payments and invoice amounts which would result in the 4090 account balance.

Can we look into why this miscalculation happened? Was there a deleted payment/invoice that did not fully update the account? I know this has happened in the past and continues to happen. I think we should be researching what is causing this, rather then manually fixing it every time.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 30, 2017, 04:57

moving to next milestone

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 10, 2017, 11:26

@smasih 

I can confirm that the manual adjustment still needs to be done, as it is showing incorrect account balances. Please manually adjust this so we are not continuing to bill the client incorrectly.

We can focus on finding the true issue behind this once priorities allow for such.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 16, 2017, 14:41

@smasih 

The manual adjustment should be a quick fix, updating priority.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Nov 17, 2017, 24:33

@kdjstudios 

Yes, the adjustments are quick but the process takes time.   
1. Firstly, we have to  take the backup and restore it on any non-production server (dev/merge/staging)
1. Then the investigation is required to identify the faulty invoice.
1. Then we have to edit all the invoices starting on from the faulty one so that it all adds up to correct balance on accounts as well as on invoices.
1. Then we do the same changes on the production server.

We will try to pick this today, @pchaudhary  will post the estimation and further update on this by eod.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 17, 2017, 07:58

@smasih 

Thank you for clarifying this is not a quick fix but a lengthy one. Please update with time estimation.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 19, 2017, 22:43

Hi [Kyle] ( @kdjstudios ),   
We dug into the issue & found that this is also the case of **incorrect balance transfer from the previous invoice.**  
After looking at the previous invoices & other information, I am quite sure that this is also because of a bug in the code previously encountered as updating the same amount of payment as the previous invoice balance.   
(*As we can see the invoice # 199-1569 should have the $ -10530.00 as the previous balance*.)  
In short - **We have to fix the previous invoices so that they can all add up to the accurate account balance, for which, we have to change the last 4 invoices.**  
Please ask Cindy to approve the changes to be done on the invoices.  
`estimation 2hr`

|

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 20, 2017, 07:58

@pchaudhary 

I have requested Cindy's approval:

The support team has followed up and found that there was a miscalculation on the invoice balance about 4 invoices ago. To correct this; the support team will have to manually adjust the last 4-5 invoices. With your approval we will instruct them to proceed in updating the invoices; Please note we would recommend to inform your client also.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 20, 2017, 08:53

Cindy has approved the changes.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Nov 20, 2017, 23:16

Hi [Kyle] ( @kdjstudios ),   
We have done the changes on following invoices. 
* # 199-1569   
* # 199-1597  
* # 199-1628  
* # 199-1693  
and in the account balance.  


The changes are live in production.  
Please ask Cindy to confirm the changes to the invoices & point out any further changes he wants ( if any).  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 21, 2017, 09:31

Thanks I am confirming with Cindy and then I will post an update.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 21, 2017, 12:46

Cindy confirmed it looks right. Closing ticket.
